bin/lint-migrations-file/src/change_set/strict.clj

```diff
@@ -28,7 +28,69 @@
       (apply distinct? (mapcat #(str/split (-> % :sql :dbms) #",")
                                changes))))))
 
+(def change-types-requiring-rollback
+  ;; this list was generated with a little grep and awk from the docs here:
+  ;; https://docs.liquibase.com/workflows/liquibase-community/liquibase-auto-rollback.html
+  #{:addAutoIncrement
+    :alterSequence
+    :createFunction
+    :createPackage
+    :createPackageBody
+    :createProcedure
+    :createTrigger
+    :customChange
+    :delete
+    :dropAllForeignKeyConstraints
+    :dropCheckConstraint
+    :dropColumn
+    :dropDefaultValue
+    :dropForeignKeyConstraint
+    :dropFunction
+    :dropIndexNot
+    :dropPackage
+    :dropPackageBody
+    :dropPrimaryKey
+    :dropProcedure
+    :dropSequence
+    :dropSynonym
+    :dropTable
+    :dropTrigger
+    :dropUniqueConstraint
+    :dropView
+    :empty
+    :executeCommand
+    :insert
+    :loadData
+    :loadUpdateData
+    :markUnused
+    :mergeColumns
+    :modifyDataType
+    :output
+    :setColumnRemarks
+    :setTableRemarks
+    :sql
+    :sqlFile
+    :stop
+    :update})
+
+(defn major-version
+  "Returns major version from id string, e.g. 44 from \"v44.00-034\""
+  [id-str]
+  (when (string? id-str)
+    (some-> (re-find #"\d+" id-str) Integer/parseInt)))
+
+(defn rollback-present-when-required?
+  "Ensures rollback key is present when change type doesn't support auto rollback"
+  [{:keys [id changes] :as change-set}]
+  (or
+   (int? id)
+   (< (major-version id) 45)
+   (not (some change-types-requiring-rollback (mapcat keys changes)))
+   (contains? change-set :rollback)))
+
 (s/def ::change-set
-  (s/merge
-   :change-set.common/change-set
-   (s/keys :req-un [::changes ::comment])))
+  (s/and
+   rollback-present-when-required?
+   (s/merge
+    :change-set.common/change-set
+    (s/keys :req-un [::changes ::comment]))))
```

Comment:
It might not be a bad idea to add some tests around this. Just to be sure we're actually catching things we think we're catching

Reply:
Done.